k8s/operators/src/pool/diskpool/client.rs

```diff
@@ -35,33 +64,149 @@ pub(crate) async fn create(
 /// is wrong would be to consult the logs.
 pub(crate) async fn ensure_crd(k8s: &Client) -> Result<CustomResourceDefinition, Error> {
     let crd_api: Api<CustomResourceDefinition> = Api::all(k8s.clone());
-
     // Check if there is an existing dsp diskpool. If yes, Replace it with new generated one.
     // Create new if it doesnt exist.
     let mut crd = DiskPool::crd();
     let crd_name = crd.metadata.name.as_ref().ok_or(Error::InvalidCRField {
         field: "diskpool.metadata.name".to_string(),
     })?;
+    crd.spec.versions[0].served = false;
```

Comment:
Do we need to check versions array rather than always assuming it's index 0?

Reply:
No crd is v1alpha1 object if you see line 69 DiskPool::crd(). There will only be 1 version in the versions array so using 0th index directly.